airflow/api_fastapi/core_api/routes/public/dag_run.py

```diff
@@ -362,16 +362,19 @@ def trigger_dag_run(
                 end=pendulum.instance(body.data_interval_end),
             )
         else:
-            data_interval = dag.timetable.infer_manual_data_interval(run_after=logical_date)
+            data_interval = dag.timetable.infer_manual_data_interval(
+                run_after=logical_date or pendulum.now("UTC")
+            )
 
-        if body.dag_run_id:
-            run_id = body.dag_run_id
-        else:
-            run_id = dag.timetable.generate_run_id(
+        run_id = (
+            body.dag_run_id
+            if body.dag_run_id
+            else dag.timetable.generate_run_id(
```

Comment:
I don't think `body.dag_run_id` can ever be none. It's automatically validated and provided a value if `if not self.dag_run_id:` via `DagRun.generate_run_id(`